CodeGeneration/Sources/generate-swift-syntax/GenerateSwiftSyntax.swift

```diff
@@ -173,13 +174,23 @@ struct GenerateSwiftSyntax: ParsableCommand {
           verbose: verbose
         )
       } catch {
+        errorsLock.lock()
         errors.append(error)
+        errorsLock.unlock()
       }
     }
 
-    if let firstError = errors.first {
-      // TODO: It would be nice if we could emit all errors
-      throw firstError
+    if errors.count > 0 {
+      struct ComposititeError: Error, CustomStringConvertible {
```

Comment:
```suggestion
      struct CompositeError: Error, CustomStringConvertible {
```

Personally I'd define this top level, probably at the bottom of this file.